src/base.ts

```diff
@@ -70,7 +70,15 @@ export class Base extends EventEmitter {
           username: this._tokenId,
           password: this._tokenSecret,
         },
-      });
+      };
+
+      if (this.config.platform?.name) {
+        request.headers[
+          'x-source-platform'
+        ] = `${this.config.platform?.name} | ${this.config.platform?.version}`;
```

Comment:
Should we also regex it out of those two parameters just to be safe?

Reply:
I think in this case, it would be better to just throw an error if either `platform.name` or `platform.version` contain a `|`.  It would be worse for us to modify a value unexpectedly.

Reply:
Okay, I added checks and clear messaging around this validation.  Also added some unit tests as well.